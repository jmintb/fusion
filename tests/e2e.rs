use anyhow::Result;
use assert_cmd::Command;
use rstest::rstest;
use std::path::PathBuf;

#[rstest]
fn test_well_formed_programs(#[files("./test_programs/*.ts")] path: PathBuf) -> Result<()> {
    let mut cmd = Command::cargo_bin("typed_script")?;
    let assert = cmd.arg("run").arg(path.clone()).assert();
    assert.success();
    insta::assert_snapshot!(
        format!(
            "test_well_formed_programs_{}",
            path.file_name().unwrap().to_str().unwrap()
        ),
        String::from_utf8(cmd.ok().unwrap().stdout)?
    );
    Ok(())
}

#[rstest]
fn snapshop_mlir_output(#[files("./test_programs/*.ts")] path: PathBuf) -> Result<()> {
    let mut cmd = Command::cargo_bin("typed_script")?;
    let assert = cmd
        .args(&["build", "--emit-mlir"])
        .arg(path.clone())
        .assert();
    assert.success();
    insta::assert_snapshot!(
        format!(
            "test_mlir_snapshot_{}",
            path.file_name().unwrap().to_str().unwrap()
        ),
        String::from_utf8(cmd.ok().unwrap().stdout)?
    );
    Ok(())
}

#[rstest]
fn snapshop_ast_output(#[files("./test_programs/*.ts")] path: PathBuf) -> Result<()> {
    let mut cmd = Command::cargo_bin("typed_script")?;
    let assert = cmd
        .args(&["build", "--emit-ast"])
        .arg(path.clone())
        .assert();
    assert.success();
    insta::assert_snapshot!(
        format!(
            "test_ast_snapshot_{}",
            path.file_name().unwrap().to_str().unwrap()
        ),
        String::from_utf8(cmd.ok().unwrap().stdout)?
    );
    Ok(())
}
