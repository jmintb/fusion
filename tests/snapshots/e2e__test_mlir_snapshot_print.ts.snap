---
source: tests/e2e.rs
expression: "String::from_utf8(cmd.ok().unwrap().stdout)?"
---
module attributes {llvm.data_layout = ""} {
  llvm.func @printf(!llvm.ptr, ...) -> i32 attributes {llvm.emit_c_interface, sym_visibility = "private"}
  llvm.mlir.global internal @annonstr0("hello world!") {addr_space = 0 : i32}
  llvm.func @main() attributes {llvm.emit_c_interface} {
    %0 = llvm.mlir.addressof @annonstr0 : !llvm.ptr
    llvm.call @print(%0) : (!llvm.ptr) -> ()
    llvm.return
  }
  llvm.func @_mlir_ciface_main() attributes {llvm.emit_c_interface} {
    llvm.call @main() : () -> ()
    llvm.return
  }
  llvm.func @print(%arg0: !llvm.ptr) {
    %0 = llvm.call @printf(%arg0) : (!llvm.ptr) -> i32
    llvm.return
  }
}


